tfx_addons/predictions_to_bigquery/utils.py

```diff
@@ -184,64 +174,90 @@ def convert_tensorflow_dtype_to_bq_type(tf_dtype: tf.dtypes.DType) -> str:
     raise ValueError(f"Unsupported type: {tf_dtype}")
 
 
-def feature_to_bq_schema(features: Dict[str, Any],
-                         required: bool = True) -> List[Dict]:
-  """
-    Convert a list of features to a list of BigQuery schema fields.
+def get_bq_field_name_from_key(key: str) -> str:
+  field_name = _REGEX_CHARS_TO_REPLACE.sub('_', key)
+  return re.sub('_+', '_', field_name).strip('_')
 
-    Args:
-        features: A list of features.
-        required: Whether the field is required.
 
-    Returns:
-        A list of BigQuery schema fields.
-    """
+def _feature_spec_to_bq_schema_fields(feature_spec: FeatureSpec,
+                                      required: bool = True) -> List[Dict]:
+  """Convert a TensorFlow feature spec to a list of BigQuery schema fields.
+
+  Args:
+    feature_spec: TensorFlow feature spec.
+    required: Whether the field is required.
+
+  Returns:
+      A list of BigQuery schema fields.
+  """
   return [{
-      "name": feature_name,
-      "type": convert_tensorflow_dtype_to_bq_type(feature_def.dtype),
+      "name": get_bq_field_name_from_key(feature_name),
+      "type": _convert_tensorflow_dtype_to_bq_type(feature_def.dtype),
       "mode": "REQUIRED" if required else "NULLABLE",
-  } for feature_name, feature_def in features.items()]
+  } for feature_name, feature_def in feature_spec.items()]
 
 
-def create_annotation_fields(
-    label_field_name: str = "category_label",
-    score_field_name: str = "score",
+def _create_annotation_fields(
```

Comment:
There's a lot of wonkiness here.
- The function takes an arbitrary number of unnamed fields to start but does nothing with them.
- Docstring looks out of date. 
- score_field_name not used.
- nothing uses add_datetime_field that calls this method.

Your prerogative on what's worth cleaning up.